### ISSUE TITLE
color mapping shapes has issues if data is a single np.array

### ISSUE THREAD

**Issue by NHPatterson** [napari/napari#1730]
## 🐛 Bug

It appears that color mapping shapes has issues if shape is a single ```np.array```. 

## To Reproduce

```python
import napari
import numpy as np

with napari.gui_qt():
    viewer = napari.Viewer()
    triangle = np.array([[11, 13], [111, 113], [22, 246]])
    viewer.add_shapes(triangle, shape_type="polygon", face_color="red")
```
## Expected behavior

Shape face color should be red. However it is white and a ```UserWarning``` is displayed:
```python
..\napari\layers\utils\color_transformations.py:50: UserWarning: The provided edge_color parameter has 3 entries, while the data contains 1 entries. Setting edge_color to white.
```
## What does work

Adding single shape as a list does work.

```python
import napari
import numpy as np

with napari.gui_qt():
    viewer = napari.Viewer()
    triangle = np.array([[11, 13], [111, 113], [22, 246]])
    viewer.add_shapes([triangle], shape_type="polygon", face_color="red")
```

## Where the problem seems to be

https://github.com/napari/napari/blob/b61d9ae570e30091a97b6c76e37cd95fe5b296b6/napari/layers/shapes/shapes.py#L1537

This line when initializing shapes checks the length of the shape. If it is a numpy array of nD coordinates, it's length will be the number of coordinates.

Changing the code to this fixes the issue. 
```python
if isinstance(data, list):
    n_shapes = len(data)
else:
    n_shapes = 1
```

Happy to open a PR on this or you can put this into an existing PR where convenient. I haven't checked but other layer types may have a similar issue.

## Environment
napari: 0.3.8
Platform: Windows-10-10.0.19041-SP0
Python: 3.8.6 | packaged by conda-forge | (default, Oct 7 2020, 18:22:52) [MSC v.1916 64 bit (AMD64)]
Qt: 5.14.2
PyQt5: 5.14.2
NumPy: 1.19.2
SciPy: 1.5.2
Dask: 2.30.0
VisPy: 0.6.5

GL version: 4.6.0 - Build 26.20.100.7325
MAX_TEXTURE_SIZE: 16384

Plugins:
- napari_plugin_engine: 0.1.8
- svg: 0.1.4

**Comment by sofroniewn** [comment]
Opps! Sorry about this @NHPatterson and thanks for the catch!

We actually have a highly related, but not the same PR #1718 in progress which changes our code for detecting the number of shapes to be done with

```python
        if len(data) == 0:
            # If no new shapes
            n_new_shapes = 0
        elif np.array(data[0]).ndim == 1:
            # If a single array for a shape
            n_new_shapes = 1
        else:
            n_new_shapes = len(data)
````

It might be nice to pull that out into a simple function

```python
def get_number_of_shapes(data):
        if len(data) == 0:
            # If no new shapes
            n_shapes = 0
        elif np.array(data[0]).ndim == 1:
            # If a single array for a shape
            n_shapes = 1
        else:
            n_shapes = len(data)
       return n_shapes
```
and then use that in all places in the class, including the place you flag. How does that sound to you?

If you wanted to make a PR to #1718 that makes this change and adds an additional test we can then consolidate the changes there and close this issue like that.

I think @kevinyamauchi wanted to do some more checking of color handling in the context #1718 so this seems very relevant there too.

**Comment by sofroniewn** [comment]
Fixed by #1718
